### PR TITLE
Update Aria2 to 1.37

### PR DESCRIPTION
**Pull request by evetion**
And hopefully link to libmbedcrypto.so.7, instead of the .5 for the current jll, which fails on current Julia releases.

### CODE REVIEW

**Comment by evetion** [comment]
Not sure what's happening on mac, did the build tools get updated since the last time? The previous build worked fine, and this also happens if I don't update aria2 to 1.37 locally, indicating something changed at our side?

Specifically we're getting [linking errors](https://buildkite.com/julialang/yggdrasil/builds/7495#018ccb5b-6ed7-4c25-8b82-5e4edd4b3029/620-4428):
```bash
[18:12:54] ld64.lld: error: undefined symbol: main
[18:12:54] >>> referenced by the entry point
[18:12:54] 
[18:12:54] ld64.lld: error: undefined symbol: __cxa_atexit
[18:12:54] >>> referenced by A2STR.cc:0
[18:12:54] >>>               .libs/A2STR.o:(symbol _GLOBAL__sub_I_A2STR.cc+0x30)
[18:12:54] >>> referenced by AuthConfigFactory.cc:0
[18:12:54] >>>               .libs/AuthConfigFactory.o:(symbol _GLOBAL__sub_I_AuthConfigFactory.cc+0x8d)
[18:12:54] >>> referenced by AuthConfigFactory.cc:0
[18:12:54] >>>               .libs/AuthConfigFactory.o:(symbol _GLOBAL__sub_I_AuthConfigFactory.cc+0x4c)
[18:12:54] >>> referenced 161 more times
[18:12:54] 
[18:12:54] ld64.lld: error: undefined symbol: std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>::~basic_string()
[18:12:54] >>> referenced by A2STR.cc:0
[18:12:54] >>>               .libs/A2STR.o:(symbol _GLOBAL__sub_I_A2STR.cc+0x23)
[18:12:54] >>> referenced by AuthConfigFactory.cc:0
[18:12:54] >>>               .libs/AuthConfigFactory.o:(symbol _GLOBAL__sub_I_AuthConfigFactory.cc+0x3a)
[18:12:54] >>> referenced by ChunkedDecodingStreamFilter.cc:0
[18:12:54] >>>               .libs/ChunkedDecodingStreamFilter.o:(symbol _GLOBAL__sub_I_ChunkedDecodingStreamFilter.cc+0x65)
[18:12:54] >>> referenced 30 more times
[18:12:54] 
[18:12:54] ld64.lld: error: undefined symbol: operator delete(void*)
...
```

File: A/Aria2/build_tarballs.jl [hunk 1]
```diff
@@ -50,11 +50,11 @@ dependencies = [
     # `LibSSH2_jll` was compiled with.  So we explicitly select the
     # right version here.
     BuildDependency(PackageSpec(name="MbedTLS_jll", version=v"2.28")),
-    Dependency(PackageSpec(name="OpenSSL_jll"); compat = "1.1.10"),
+    Dependency(PackageSpec(name="OpenSSL_jll"); compat="1.1.10"),
```
Comment (giordano) [inline A/Aria2/build_tarballs.jl:53]: It's better to use OpenSSL 3 at this point
```suggestion
    Dependency(PackageSpec(name="OpenSSL_jll"); compat="3.0.8"),
```

File: A/Aria2/build_tarballs.jl [hunk 2]
```diff
@@ -50,11 +50,11 @@ dependencies = [
...
     Dependency(PackageSpec(name="XML2_jll")),
     Dependency(PackageSpec(name="Zlib_jll")),
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
 build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
-               julia_compat="1.6", preferred_gcc_version = v"7")
+    julia_compat="1.6", preferred_gcc_version=v"7")
```
Comment (giordano) [inline A/Aria2/build_tarballs.jl:60]:
```suggestion
               julia_compat="1.8", preferred_gcc_version=v"7")
```

File: A/Aria2/build_tarballs.jl [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
 
 name = "Aria2"
-version = v"1.36.0"
+version = v"1.37.0"
 
 # Collection of sources required to complete build
 sources = [
     ArchiveSource("https://github.com/aria2/aria2/releases/download/release-$(version)/aria2-$(version).tar.xz",
-                  "58d1e7608c12404f0229a3d9a4953d0d00c18040504498b483305bcb3de907a5"),
+        "60a420ad7085eb616cb6e2bdf0a7206d68ff3d37fb5a956dc44242eb2f79b66b"),
```
Comment (giordano) [inline A/Aria2/build_tarballs.jl:11]:
```suggestion
                  "60a420ad7085eb616cb6e2bdf0a7206d68ff3d37fb5a956dc44242eb2f79b66b"),
```

File: A/Aria2/build_tarballs.jl (hunk 2 above)
Comment (evetion) [reply A/Aria2/build_tarballs.jl]: Thank you, I missed the `clang_use_lld` option.

**Comment by evetion** [comment]
Hmm it seems the CI ran out of disk space? :/

**Pull request merged**